docs/rules/no-undefined.md

```diff
@@ -14,24 +14,15 @@ function doSomething(data) {
 }
 ```
 
-This represents a problem for `undefined` that doesn't exist for `null`, which is a keyword and primitive value that can neither be overwritten nor shadowed.
+Because `undefined` can be overwritten or shadowed, reading `undefined` can give an unexpected value. (This is not the case for `null`, which is a keyword that always produces the same value.) To guard against this, you can use the [no-global-assign](no-global-assign.md) and [no-shadow-restricted-names](no-shadow-restricted-names.md) rules.
```

Comment:
I like the changes you make before "To guard against this"; however, I think it would make more sense to extract the rest to its own paragraph and make it clear that that is an alternative to using this rule to flag *all* uses of `undefined`. (And I would put that paragraph below the next one-- see comment below.)

See comment below